zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/CallActivityTransformer.java

```diff
@@ -53,4 +57,19 @@ private void transformProcessId(
         calledElement.isPropagateAllParentVariablesEnabled();
     callActivity.setPropagateAllParentVariablesEnabled(propagateAllParentVariablesEnabled);
   }
+
+  private static void transformLexicographicIndex(
+      final CallActivity element,
+      final List<ExecutableProcess> processes,
+      final ExecutableCallActivity callActivity) {
+    final List<String> allCallActivityIds =
+        processes.stream()
+            .flatMap(p -> p.getFlowElements().stream())
```

Comment:
❓ This appears to be collecting all elements of **all deployed processes**, but you previously mentioned to me that only a single process is available. Did that concern this place? If so, let's add a comment to this code clarifying that this builds the lexicographic index within the process, not within all processes. 

💭 Alternatively, we could also only map the first entry, rather than flatmap all processes to clarify that there only is one that is relevant. Still this requires a clarifying comment.

Reply:
To clarify, I previously meant that we only have processes from one deployment resource. So if we deployed multiple processes in one file, then we have access to all of them. And that's why I'm collecting from all deployed processes here.

Check the new test case added [here](https://github.com/camunda/camunda/pull/20293/commits/31273c61efd61cbf01792acd8afe98ba53b7f95b)

Reply:
👍 Ah okay. Thanks for clarifying. 

🔧 Let's make this clear in the JavaDoc of the `lexicographicIndex` field. It should also be clear to Operate that this is how it works.